### PR TITLE
PSR-507 Loan & Land or Property - User allowed to navigate to List pa…

### PR DESCRIPTION
**Pull request by cyeniceri**
…ge without completing the loan/Land journey

### CODE REVIEW

File: app/config/FrontendAppConfig.scala [hunk 1]
```diff
@@ -20,13 +20,8 @@ import com.google.inject.{Inject, Singleton}
...
 import play.api.i18n.Lang
 import play.api.mvc.RequestHeader
-import uk.gov.hmrc.play.bootstrap.binders.{
-  AbsoluteWithHostnameFromAllowlist,
-  OnlyRelative,
-  RedirectUrl,
-  SafeRedirectUrl
-}
 import uk.gov.hmrc.play.bootstrap.binders.RedirectUrl._
+import uk.gov.hmrc.play.bootstrap.binders.{AbsoluteWithHostnameFromAllowlist, OnlyRelative, RedirectUrl}
```
Comment (kberec) [inline app/config/FrontendAppConfig.scala:24]: Why are these needed?

File: build.sbt [hunk 2]
```diff
@@ -10,7 +10,7 @@ lazy val appName: String = "pension-scheme-return-frontend"
 addCompilerPlugin("org.typelevel" % "kind-projector" % "0.13.2" cross CrossVersion.full)
 
 lazy val root = (project in file("."))
-  .enablePlugins(PlayScala, SbtAutoBuildPlugin, SbtDistributablesPlugin)
+  .enablePlugins(PlayScala, SbtDistributablesPlugin)
```
Comment (kberec) [inline build.sbt:13]: Why do we need this?

File: build.sbt [hunk 3]
```diff
@@ -96,7 +96,7 @@ lazy val root = (project in file("."))
   )
 
 lazy val testSettings: Seq[Def.Setting[?]] = Seq(
-  fork := false,
+  fork := true,
```
Comment (kberec) [inline build.sbt:99]: Why do we need this?

File: build.sbt [hunk 4]
```diff
@@ -109,5 +109,5 @@ lazy val itSettings = Defaults.itSettings ++ Seq(
     baseDirectory.value / "it" / "resources"
   ),
   parallelExecution := false,
-  fork := false
+  fork := true
```
Comment (kberec) [inline build.sbt:112]: Why do we need this?

File: app/controllers/nonsipp/landorpropertydisposal/LandOrPropertyDisposalListController.scala [hunk 5]
```diff
@@ -54,12 +55,11 @@ class LandOrPropertyDisposalListController @Inject()(
 
   def onPageLoad(srn: Srn, page: Int, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) {
     implicit request =>
-      getDisposals(srn).map { disposals =>
-        val numberOfDisposal = disposals.map { case (_, disposalIndexes) => disposalIndexes.size }.sum
-        if (numberOfDisposal == 0) {
-          Redirect(routes.LandOrPropertyDisposalController.onPageLoad(srn, NormalMode))
-        } else {
+      val (status, incompleteDisposalUrl) = getDisposalsTaskListStatusWithLink(request.userAnswers, srn)
```
Comment (kberec) [inline app/controllers/nonsipp/landorpropertydisposal/LandOrPropertyDisposalListController.scala:58]: This didn't work for me when I was doing a disposal the second time for the first land or property. It didn't detect the incomplete disposal

File: build.sbt (hunk 4 above)
Comment (cyeniceri) [reply build.sbt:112]: reverted

File: build.sbt (hunk 3 above)
Comment (cyeniceri) [reply build.sbt:99]: reverted

File: build.sbt (hunk 2 above)
Comment (cyeniceri) [reply build.sbt:13]: https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?spaceKey=TEC&postingDay=2021%2F5%2F5&title=New+version+of+sbt-auto-build
we don't need to enable this, it's auto enabled

File: app/config/FrontendAppConfig.scala (hunk 1 above)
Comment (cyeniceri) [reply app/config/FrontendAppConfig.scala:24]: reverted
